client/src/pages/datasets/datasetContentsView/[datasetName].tsx

```diff
@@ -0,0 +1,92 @@
+import {useAuth} from 'contexts/auth';
+import {getFilesFromDataset} from 'lib/api/files';
+import React, {useEffect, useState} from 'react';
+import {UserFile} from 'types/UserFile';
+import {useRouter} from 'next/router';
+import Prose from 'components/Prose';
+
+export default function DatasetViewer() {
+  const router = useRouter();
+  const datasetName = router.query.datasetName as string;
```

Comment:
I'd use object destructuring here:
```
const {datasetName} = router.query
```
You might not need the typescript cast here but I may be wrong